source/mir/ndslice/algorithm.d

```diff
@@ -1995,3 +1997,942 @@ size_t countImpl(alias fun, Slices...)(Slices slices)
+    //| 4 5 6 |
+    //| 7 8 9 |
+    auto m = iota([3, 3], 1).slice;
+    m.eachUpper!((ref a) {a = 0; });
+    assert(m == [
+        [1, 0, 0],
+        [4, 5, 0],
+        [7, 8, 9]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1 2 3 |
+    //| 4 5 6 |
+    //| 7 8 9 |
+    auto m = iota([3, 3], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(-1);
+    assert(m == [
+        [0, 0, 0],
+        [0, 0, 0],
+        [7, 0, 0]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1 2 3 |
+    //| 4 5 6 |
+    //| 7 8 9 |
+    auto m = iota([3, 3], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(2);
+    assert(m == [
+        [1, 2, 0],
+        [4, 5, 6],
+        [7, 8, 9]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1 2 3 |
+    //| 4 5 6 |
+    //| 7 8 9 |
+    auto m = iota([3, 3], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(-2);
+    assert(m == [
+        [0, 0, 0],
+        [0, 0, 0],
+        [0, 0, 0]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1  2  3  4 |
+    //| 5  6  7  8 |
+    //| 9 10 11 12 |
+    auto m = iota([3, 4], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(0);
+    assert(m == [
+        [0, 0, 0, 0],
+        [5, 0, 0, 0],
+        [9, 10, 0, 0]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1  2  3  4 |
+    //| 5  6  7  8 |
+    //| 9 10 11 12 |
+    auto m = iota([3, 4], 1).slice;
+    m.eachUpper!((ref a) {a = 0; });
+    assert(m == [
+        [1, 0, 0, 0],
+        [5, 6, 0, 0],
+        [9, 10, 11, 0]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1  2  3  4 |
+    //| 5  6  7  8 |
+    //| 9 10 11 12 |
+    auto m = iota([3, 4], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(-1);
+    assert(m == [
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [9, 0, 0, 0]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1  2  3  4 |
+    //| 5  6  7  8 |
+    //| 9 10 11 12 |
+    auto m = iota([3, 4], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(2);
+    assert(m == [
+        [1, 2, 0, 0],
+        [5, 6, 7, 0],
+        [9, 10, 11, 12]]);
+}
+
+pure nothrow
+version(mir_test) unittest
+{
+    import mir.ndslice.allocation: slice;
+    import mir.ndslice.topology: iota;
+
+    //| 1  2  3  4 |
+    //| 5  6  7  8 |
+    //| 9 10 11 12 |
+    auto m = iota([3, 4], 1).slice;
+    m.eachUpper!((ref a) {a = 0; })(-2);
```

Comment:
`m.eachUpper!"a = 0"` should work too. Please use it since it do not generate template bloat for each difinition.